myst_nb/parser.py

```diff
@@ -112,6 +113,33 @@ def parse_block(src, start_line):
                 dup_ref["fixed"] = True
         return tokens
 
+    def parse_code_cell(cell, start_line):
+        tokens = [
+            Token(
+                "nb_code_cell",
+                "",
+                0,
+                meta={"cell": cell},
+                map=[start_line, start_line],
+            )
+        ]
+        for i, output in enumerate(cell["outputs"]):
+            if output["output_type"] == "display_data":
+                if "text/markdown" in output["data"]:
+                    new_code_cell = deepcopy(cell)
+                    new_code_cell["metadata"]["tags"] = new_code_cell["metadata"].get(
+                        "tags", []
+                    ) + ["remove-input"]
```

Comment:
Wouldn't we want to create a new *markdown* cell instead of a new code cell? Then not worry about outputs etc because there wouldn't be any for a markdown cell

Reply:
what this is saying is to split code cells on markdown displays. it is possible that there is a text/html or widget following a markdown output and we want to process them.

Reply:
L136 is the markdown cell you are referring to. the new code cell is the output content potentially following the markdown.

Reply:
> it is possible that there is a text/html or widget following a markdown output and we want to process them.

this behaviour should be "codified" into a test(s), to demonstrate the behaviour, and check that it is working as expected

Reply:
where is a good place to put that test?

chris, could you please try to help me along with as i make these first contributions.  these projects have a lot of content and i need help getting familiar. i'd like to help y'all and i could use a little hand holding.

Reply:
>  i'd like to help y'all and i could use a little hand holding.

I'll try to help where possible, but unfortunately I don't have a large amount of time to devote to this at present. I appreciate that this is your first contribution, and that's great and I encourage it. 
But at the same time this is not a trivial piece of code being added, and I want to make sure that it is not going to introduce new bugs for all the other users, which it appears to do at the moment.
This first round of review is about identifying potential issues, then we can work on fixing them.

Reply:
could you suggest where you would like this test. is it a standalone test? does it go with an existing test? what bugs am i testing for? 

i'm neurodivergent and i am not good at implicitly understanding what people mean. if you're too busy then maybe i could walk through the review process with another contributor?

Reply:
> it is possible that there is a text/html or widget following a markdown output and we want to process them.

Add a notebook in the tests folder, that has a text/html or widget following a markdown output, then copy one of the other tests to run it and test that the output is as expected 😄 

Then add to the notebook additional "scenarios" of different markdown content, and different cell outputs (markdown then stdout, stdout then markdown, ...)

I'm sure @choldgraf can guide through the rest